src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,415 @@
+from ..techniques.ms import MSSpectrum
+from ..spectra import Spectrum, SpectrumSeries
+from ..config import plugins
+
+SCALE_TIME_TO_SECONDS = 1e-3
+
+
+class CinfdataDBReader:
+    """A class that connects to cinf_database or read from cache
+    https://cinfdata-dababase-client.readthedocs.io/en/latest/index.html
+
+    Attributes:
+        setup_name (str): The setup name in the DB
+        timestamp (str): Timestamp when the experiment started in YYYY-MM-DD HH:MM:SS
+        units (dict): Dictionary of columns names with corresponding units
+        tstamp (str): The unix time corresponding to t=0 for the measurement
+        technique (str): The name of the technique
+        measurement (Measurement): The measurement returned by read() when the database
+            is read. self.measurement is None before read() is called.
+    """
+
+    def __init__(self):
+        """Initialize a Reader for cinf_database. See class docstring."""
+        self.name = None
+        self.sample_name = None
+        self.setup_name = None
+        self.timestamp = None
+        self.tstamp = None
+        self.data_has_been_fetched = False
+        self.metadata = {}
+        self.technique = "reactor"
+        self.measurement_class = None
+        self.measurement = None
+        self.cinf_db = None
+        self.mass_scans = False
+        self.spectrum_list = []
+        plugins.activate_cinfdata()
+
+    def read(self, path_to_file, name=None, cls=None, units=None, **kwargs):
+        """
+        Return a xx-Measurement or Spectrum with the data and metadata recorded from
+        a setup at SurfCat at given timestamp
+        All attributes of this reader can be accessed from the
+        measurement as `measurement.reader.attribute_name`.
+
+        Args:
+        path_to_file (str): Named argument from Measurement Class.
+            Can be used as the setup name in the cinfdatabase
+        **kwargs (dict): Key-word arguments are passed to cinf Measurement.__init__
+            setup_name (str): The setup name in the database default to path_to_file
+            timestamp (str): Timestamp the measurement started given as
+                (YYYY-MM-DD HH:MM:SS)
+        """
+
+        if self.data_has_been_fetched:
+            print(
+                f"This {self.__class__.__name__} has already fetched data from "
+                f" {self.token} grouped by {self.grouping_column}"
+                " Returning the measurement resulting from the original read. "
+                "Use a new Reader if you want to read another file."
+            )
+
+            return self.measurement
+
+        self.measurement_class = kwargs.pop("measurement_class", cls)
+        self.setup_name = kwargs.pop("setup_name", path_to_file)
+        self.timestamp = kwargs.pop("timestamp", None)
+        self.comment = kwargs.pop("comment", None)
+        self.mass_scans = kwargs.pop("include_mass_scans", False)
+        self.verbose = kwargs.pop("verbose", False)
```

Comment:
Need a `self.verbose = False` in `__init__`!

Reply:
Good point.

The attributes are now set in init with a relevenat value and the rest is made local variables. ( hopefully ).

Reply:
Looks good!